src/api.jl

```diff
@@ -50,33 +50,70 @@ function Base.show(io::IO, e::HTTPError)
     end
 end
 
+"""
+    wait_until_done(ctx::Context, rsp::TransactionResponse)
+    wait_until_done(ctx::Context, transaction)
+    wait_until_done(ctx::Context, txn_id::String)
+
+Block until the provided `transaction` has reached a terminal state.
+
+Continuously polls get_transaction() for the transaction's state, until the transaction has
+finished. A transaction has finished once it has reached one of the terminal states:
+`COMPLETED` or `ABORTED`. The polling uses a low-overhead exponential backoff in order to
+ensure low-latency results without overloading network traffic.
+"""
+function wait_until_done(ctx::Context, rsp::TransactionResponse)
+    wait_until_done(ctx, rsp.transaction)
+end
+function wait_until_done(ctx::Context, txn::JSON3.Object)
+    wait_until_done(ctx, transaction_id(txn))
+end
+function wait_until_done(ctx::Context, id::AbstractString)
+    try
+        _poll_with_specified_overhead(overhead_rate = 0.01) do
+            txn = get_transaction(ctx, id)
+            return transaction_is_done(txn)
+        end
+        t = @spawn get_transaction(ctx, id)
+        m = @spawn get_transaction_metadata(ctx, id)
+        p = @spawn get_transaction_problems(ctx, id)
+        r = @spawn get_transaction_results(ctx, id)
+
+        return TransactionResponse(fetch(t), fetch(m), fetch(p), fetch(r))
+    catch
+        # Always print out the transaction id so that users can still get the txn ID even
+        # if there's an error during polling (such as an InterruptException).
+        @error "Client-side error while executing transaction." transaction_id=id
+        rethrow()
+    end
+end
+
 # Polls until the execution `f()` is truthy or the maximum number of polls is
-# reached. Polling frequency is controlled by an `ExponentialBackOff`. If `throw_on_max_n`
-# is set to true, this will throw if the maximum number of iterations are reached.
-function _poll_until(
+# reached. Polling frequency is controlled to minimize overhead, by keeping the overhead
+# from sleeping within a specified overhead rate. If `throw_on_max_n` is set to true, this
```

Comment:
@bachdavi: Hopefully this is somewhat clearer?:
> # Polling frequency is computed to minimize overhead: we carefully set the sleep time
> # between polls to a fraction of the time waited so far, so that after any given sleep, we
> # cannot have missed the upstream result by more than `overhead_rate` of the _actual time_.